Lib/inspect.py

```diff
@@ -739,6 +739,10 @@ def getdoc(object):
             return None
     if not isinstance(doc, str):
         return None
+
+    from dataclasses import is_dataclass, add_field_docs
+    if is_dataclass(object):
+        doc = add_field_docs(object, doc)
```

Comment:
Perhaps this could go into `_finddoc()` (under `if isclass(obj):`)?
E.g.:
```diff
diff --git a/Lib/inspect.py b/Lib/inspect.py
index 7e9f7ce11e..e788dc60e9 100644
--- a/Lib/inspect.py
+++ b/Lib/inspect.py
@@ -662,6 +662,7 @@ def _findclass(func):
 
 def _finddoc(obj):
     if isclass(obj):
+        doc = None
         for base in obj.__mro__:
             if base is not object:
                 try:
@@ -669,8 +670,10 @@ def _finddoc(obj):
                 except AttributeError:
                     continue
                 if doc is not None:
-                    return doc
-        return None
+                    break
+        if is_dataclass(obj):
+            doc = add_field_docs(obj, doc or '')
+        return doc
 
     if ismethod(obj):
         name = obj.__func__.__name__
```